wcfsetup/install/files/lib/system/mail/Mail.class.php

```diff
@@ -327,10 +221,26 @@ public function getMessage() {
 	 */
 	public function setFrom($from) {
 		if (is_array($from)) {
-			$this->from = self::buildAddress(key($from), current($from));
+			$this->email->setSender(new Mailbox(current($from), key($from)));
+		}
+		else {
+			$this->email->setSender(new Mailbox($from));
```

Comment:
This change is bogus. `$from` might be an address specification in `name-addr` format.